modules/apps/headless/headless-discovery/headless-discovery-test/src/testFunctional/macros/APIExplorer.macro

```diff
@@ -127,6 +127,98 @@ definition {
 			method = ${method});
 	}
 
+	@summary = "Default summary"
+	macro executeAPIMethodCompact(api = null, version= null, parameter_two = null, parameter_three = null, parameter_four = null, method = null, requestBody = null, service = null, parameter = null, scopeKey = null, value_two = null, direction = null, assertValue = null) {
+		APIExplorer.navigateToOpenAPI(
+			api = ${api},
+			version = ${version});
+
+		Variables.assertDefined(parameterList = "${service},${method}");
+
+		APIExplorer.expandAPIMethod(
+			method = ${method},
+			service = ${service});
+
+		if (isSet(parameter)) {
+			if (${parameter} == "siteId") {
+				if (!(isSet(groupName))) {
+					var groupName = "Guest";
+				}
+
+				var value = JSONGroupAPI._getGroupIdByNameNoSelenium(
+					groupName = ${groupName},
+					site = "true");
+			}
+
+			Type(
+				locator1 = "OpenAPI#PARAMETER_UNDER_METHOD",
+				method = ${method},
+				parameter = ${parameter},
+				value1 = ${value});
+
+			if (isSet(parameter_two)) {
```

Comment:
Can we use better names for these parameters?

Reply:
Sadly no, as the parameters can differ for every endpoint.